nextflow_schema.json

```diff
@@ -479,6 +479,82 @@
+                },
+                "bamfiltering_generatemappedfastq": {
+                    "type": "boolean",
+                    "description": "Generate FASTQ files containing only mapped reads from the aligner generated BAM files .",
+                    "help_text": "This turns on the generation and saving in the results directory of FASTQs of only the mapped reads from the mapping step, using `samtools fastq`.\n\nThis could be useful if you wish to do other analysis of the mapped reads independently of the pipeline, such as remapping with different parameters (whereby only including mapped reads will speed up computation time during the re-mapping due to reduced input data).\n\nNote the reads in these FASTQ files have _not_ undergone length of quality filtering\n\n> Modifies tool parameter(s):\n> - samtools fastq: `-F 4`",
+                    "fa_icon": "far fa-file-alt"
+                },
+                "bamfiltering_retainunmappedgenomicbam": {
+                    "type": "boolean",
+                    "description": "Specify to retain unmapped reads in the BAM file used for downstream genomic analyses.",
+                    "help_text": "You can use this parameter to retain unmapped reads (optionally also length filtered) in the genomic BAM for downstream analysis - as by default, the pipeine only keeps mapped reads for downstream.\n\nThis is also turned on if `--metagenomicscreening_input` is set to all.\n\n> ⚠️ This will likely slow down run time of downstream pipeline steps!\n\n> Modifys tool parameter(s):\n> - samtools view: `-f 4` / `-F 4`",
+                    "fa_icon": "fas fa-piggy-bank"
+                },
+                "bamfiltering_savefilteredbams": {
+                    "type": "boolean",
+                    "description": "Save in the results directory the intermediate filtered genomic BAM files that are sent for  downstream genomic analyses.",
+                    "help_text": "This saves intermediate length and/or quality filtered genomic BAM files in the results directory. ",
+                    "fa_icon": "far fa-save"
+                }
+            },
+            "fa_icon": "fas fa-filter"
+        },
+        "metagenomics": {
+            "title": "Metagenomics",
+            "type": "object",
+            "description": "Options to related to metagenomic screening.",
+            "default": "",
+            "properties": {
+                "run_metagenomicscreening": {
+                    "type": "boolean",
+                    "description": "Turn on metagenomic screening of mapped, unmapped, or all reads.",
+                    "fa_icon": "fas fa-power-off",
+                    "help_text": "Turns on the metagenomic screening subworkflow of the pipeline, where reads are screening against large databases. Typically used for pathogen screening or microbial community analysis.\n\nIf supplied, this will also turn on the BAM filtering subworkflow of the pipeline."
```

Comment:
```suggestion
                    "help_text": "Turns on the metagenomic screening subworkflow of the pipeline, where reads are screened against large databases. Typically used for pathogen screening or microbial community analysis.\n\nIf supplied, this will also turn on the BAM filtering subworkflow of the pipeline."
```